tests/test_examples.py

```diff
@@ -34,39 +36,69 @@ def get_file_paths():
     return test_files
 
 
-def create_checker(test_file, checker_name):
-    """Creates a test function from a test file, and a checker name.
-    test_file: The full path (string) to the file.
-    checker_name: The hyphenated checker name that should be detected.
-    An example of a valid checker_name would be: 'no-init-classes'
+def run_pylint_on_examples() -> Dict[str, str]:
+    """Return a dict mapping example file path to its pylint output
     """
-    # The following are captured when this function is created.
-    def new_test_func():
+
+    pylint_output = {}
+
+    file_paths = get_file_paths()
+    full_output = _get_full_pylint_output(file_paths)
+    _, *individual_outputs = full_output.split("*************")
+    for output, file_path in zip(individual_outputs, file_paths):
+        _assert_parallel_output(output, file_path)
+        pylint_output[file_path] = output
+
+    return pylint_output
+
+
+def _assert_parallel_output(output: str, file_path: str) -> None:
+    """Assert that the output is for the given file"""
+    file_base_name = os.path.basename(file_path)
+    file_name, _ = os.path.splitext(file_base_name)
+    assert file_name in output
+
+
+def _get_full_pylint_output(file_paths: List[str]) -> str:
+
+    output = subprocess.run(
+        ['pylint', '--reports=n',
+         '--rcfile=python_ta/.pylintrc',
+         *file_paths],
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE)
+
+    return output.stdout.decode('UTF-8')
+
+
+class TestExamples:
```

Comment:
You don't need a class here. Below, you're using a pytest fixture, which is a good approach. You can just make the fixture return the dictionary to avoid storing it in a variable, and then pass the fixture to the actual test function.

You can read more about pytest fixtures here: https://docs.pytest.org/en/6.2.x/fixture.html.